inspirehep/modules/disambiguation/core/ml/utils.py

```diff
@@ -0,0 +1,369 @@
+
+
+def get_author_full_name(s):
+    """Get author full name from the signature.
+
+    Parameters
+    ----------
+    :param s: dict
+        Signature
+
+    Returns
+    -------
+    :returns: string
+        Normalized author name
+    """
+    v = s["author_name"]
+    v = normalize_name(v) if v else ""
+    return v
+
+
+def get_first_given_name(s):
+    """Get author first given name from the signature.
+
+    Parameters
+    ----------
+    :param s: dict
+        Signature
+
+    Returns
+    -------
+    :returns: string
+        Author's first given name
+    """
+    v = given_name(s["author_name"], 0)
+    return v
+
+
+def get_second_given_name(s):
+    """Get author second given name from the signature.
+
+    Parameters
+    ----------
+    :param s: dict
+        Signature
+
+    Returns
+    -------
+    :returns: string
+        Author's second given name
+    """
+    v = given_name(s["author_name"], 1)
+    return v
+
+
+def get_surname(s):
+    return s['author_name'].split(" ")[0].split(",")[0]
+
+
+def get_first_initial(s):
+    v = given_name_initial(s["author_name"], 0)
+    try:
+        return v
+    except IndexError:
+        return ""
+
+
+def get_second_initial(s):
+    """Get author second given name's initial from the signature.
+
+    Parameters
+    ----------
+    :param s: dict
+        Signature
+
+    Returns
+    -------
+    :returns: string
+        Second given name's initial. Empty string in case it's not available.
+    """
+    v = given_name_initial(s["author_name"], 1)
+    try:
+        return v
+    except IndexError:
+        return ""
+
+
+def get_author_other_names(s):
+    """Get author other names from the signature.
+
+    Parameters
+    ----------
+    :param s: dict
+        Signature
+
+    Returns
+    -------
+    :returns: string
+        Normalized other author names
+    """
+    v = s["author_name"]
+    v = v.split(",", 1)
+    v = normalize_name(v[1]) if len(v) == 2 else ""
+    return v
+
+
+def get_author_initials(s):
+    """Get author initials from the signature.
+
+    Parameters
+    ----------
+    :param s: dict
+        Signature
+
+    Returns
+    -------
+    :returns: string
+        Initials, not separated
+    """
+    v = s["author_name"]
+    v = v if v else ""
+    v = "".join(name_initials(v))
+    return v
+
+
+def get_author_affiliation(s):
+    """Get author affiliation from the signature.
+
+    Parameters
+    ----------
+    :param s: dict
+        Signature
+
+    Returns
+    -------
+    :returns: string
+        Normalized affiliation name
+    """
+    v = s["author_affiliation"]
+    v = normalize_name(v) if v else ""
+    return v
+
+
+def get_title(s):
+    """Get publication's title from the signature.
+
+    Parameters
+    ----------
+    :param s: dict
+        Signature
+
+    Returns
+    -------
+    :returns: string
+        Title of the publication
+    """
+    v = s["publication"]["title"]
+    v = v if v else ""
+    return v
```

Comment:
Logic like this can be rewritten as `get_value(signature, 'publication.title', default='')`, but this assumes the eager join that I'd like to eliminate for the reason explained in https://github.com/inspirehep/inspire-next/pull/3455/files#r195340445.